### PR TITLE
Bugfix: LoFTR was ignoring mask input

### LINKED ISSUE #2688: Mask not applied in LoFTR
### Describe the bug

Masks are never applied in LoFTR

Here https://github.com/kornia/kornia/blob/40d07ec3aedbbdc83d7f2e613966980ae69f7bbe/kornia/feature/loftr/loftr.py#L157 it checks if `mask0` and `mask1` are in `_data`, but `_data` was never updated to include these fields from `data`

First introduced here https://github.com/kornia/kornia/pull/2092/files#diff-203f42cbe6ec8dd5b92638ea763282d0a2b33d8d35786618a84c74919d85b2adL153

### Reproduction steps

```bash
Run the following

import cv2
import numpy as np
import torch
import kornia

img0 = torch.rand(1, 1, 100, 100)
img1 = torch.rand(1, 1, 100, 100)
# allow only matching on the right 20 pixels
mask0 = torch.zeros((1, 100, 100))
mask0[:, :, 80:] = 1
mask1 = torch.zeros((1, 100, 100))
mask1[:, :, 80:] = 1
loftr = kornia.feature.LoFTR('outdoor')
out = loftr({'image0': img0, 'image1': img1, 'mask0': mask0, 'mask1': mask1})
# check that there are no matching with x value less than 80
print((out['keypoints0'][:, 0] < 80).sum())
```
```


### Expected behavior

Always output 0

### Environment

```shell
- PyTorch Version (e.g., 1.0): 2.1.0
- OS (e.g., Linux): Ubuntu 22.04
- How you installed PyTorch (`conda`, `pip`, source): pip
- Build command you used (if compiling from source):
- Python version: 3.10.6
- CUDA/cuDNN version: 12.1
- GPU models and configuration: 4090
- Any other relevant information: container nvcr.io/nvidia/pytorch:23.06-py3
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Yosshi999**
#### Changes

Fixes #2688 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by johnnv1 — APPROVED** [review]
nice catch
